src/directReservation/directReservationPage.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { Container } from 'react-grid-system';
+import { Helmet } from 'react-helmet';
+import { Button, Notification } from 'hds-react';
+
+import { openLoginModal } from '../login/actions';
+import MainContentElement from '../a11y/MainContentElement';
+import { getPageTitle } from '../root/helpers';
+import { useNavigate, useParams } from 'react-router';
+import AuthDependentContent from '../auth/components/authDependentContent';
+import { generateFavourite } from './actions';
+import { RootState } from '../root/rootReducer';
+import { AppRoutes, getRouteById } from '../root/routes';
+import BlockLoader from '../loader/blockLoader';
+
+interface State {
+  isGeneratingFavourite: boolean;
+  generationFailed: boolean;
+  generationSuccessfull: boolean;
+}
+
+interface Props {
+  isGeneratingFavourite: boolean;
+  generationFailed: boolean;
+  generationSuccessfull: boolean;
+  openLoginModal: () => void;
+  generateFavourite: (id: string) => void;
+}
+
+const DirectReservationPage = ({
+  openLoginModal,
+  generateFavourite,
+  isGeneratingFavourite,
+  generationFailed,
+  generationSuccessfull,
+}: Props): JSX.Element => {
+  const { t } = useTranslation();
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  return (
+    <AuthDependentContent>
+      {(loading, loggedIn) => {
+        useEffect(() => {
+          if (loggedIn && id) {
+            generateFavourite(id);
+          }
+
+          if (generationSuccessfull) {
+            navigate(getRouteById(AppRoutes.FAVOURITES));
+          }
+        }, [loggedIn, generationSuccessfull]);
+
+        return (
+          <MainContentElement className="DirectReservationPage">
+            <Helmet>
+              <title>
+                {getPageTitle(
+                  t('directReservation.pageTitle', 'Direct Reservation')
+                )}
+              </title>
+            </Helmet>
+            <Container className="DirectReservationPage__content">
+              {loading && <BlockLoader />}
+              {loggedIn ? (
+                <>
+                  {isGeneratingFavourite && <BlockLoader />}
+                  {generationFailed && (
+                    <Notification type="error">
+                      {t(
+                        'direactReservation.errors.favouriteGenerationError',
+                        'Something went wrong! Try again later or contact the city official you received the direct reservation link from.'
+                      )}
+                    </Notification>
+                  )}
+                </>
+              ) : (
+                <>
+                  <p>
+                    {t(
```

Comment:
This flashes briefly before the link is processed even if you are already logged in.

Reply:
hmm, I could not reproduce this? :thinking:

Reply:
Interesting, I'm seeing this every time I test this. It's even showing both the loader and the login message as long as the favourite hasn't been received and the redirect hasn't happened yet, I think.

![image](https://github.com/City-of-Helsinki/mvj-ui-public/assets/12729676/5a631425-859b-4422-8987-5164ed1e00cf)

Reply:
ahh. now when logging out and then logging in again I get this also. All kind of other wonkiness is happening there also..

Reply:
Now it is fixed. Duh, I am stupido.